content/posts/2018-12-09-abandon-jenkins.md

```diff
@@ -0,0 +1,192 @@
+things a small development team can manage on its own without losing
+the primary goal — to improve the business value of a product.
+
+Fortunately enough, talking about CI, there is a variety of options to delegate
+this task:
+[Travis](https://travis-ci.org/),
+[Circle](https://circleci.com/),
+[Bitrise](https://www.bitrise.io/) and more.
+
+* The maintenance is done not by a team itself but a dedicated unit.
+  Not the same thing as a unit in the same company though —
+  it will be necessary to adapt to the CI platform instead of adapting
+  the platform for the team. At the same time, there are benefits of a free
+  market. Not satisfied with the platform — move to another one.
+* Surprisingly it is actually cheaper than having an in-house solution.
+  Jenkins does not automatically start and stop itself. That means
+  always working server which results in whooping Amazon AWS bills.
+  Dedicated services optimize costs via starting and stopping isolated
+  containers on demand.
+
+# Bitrise
+
+It was a natural choice for the team because it is the only service that works
+with [Bitbucket Server](https://www.atlassian.com/software/bitbucket/server)
+([not the same thing as the cloud version](https://confluence.atlassian.com/confeval/development-tools-evaluator-resources/bitbucket/bitbucket-cloud-vs-server)).
+Interesting fact — Bitbucket Server support
+[was added by an open source contributor](https://github.com/bitrise-io/bitrise-webhooks/pull/67).
+It is possible because Bitrise components are [open source](https://github.com/bitrise-io).
+
+[Pricing is reasonable](https://www.bitrise.io/pricing). I like that
+the hardware configuration is not hidden away from customers.
+At the same time there is an opt-in for Elite machines with better specs.
+For some reason a lot of services hide that kind of information and do not
+allow to upgrade the hardware. I was told that Linux machines are provided
+via Google Compute Engine. Amazon instances for Jenkins costed us $1000
+a month, Bitrise Elite costs $360 for basically the same hardware.
+Plus no human resources to support Jenkins!
+
+Since Bitrise people understand that caching is important for shorter
+build times there is a multi-level configurable caching. For Android projects
+basically everything `sdkmanager` provides is bundled into the underlying
+Docker image — I haven’t seen a single issue with missing SDK packages.
+At the same time, it is possible to reuse files across builds, which can
+be adapted to include Gradle build cache, dependencies — you name it.
+
+Of course the configuration is declarative and understandable.
+There is no need to be a DevOps to read it.
+
+```yaml
+trigger_map:
+- tag: "v*"
+  workflow: release
+
+workflows:
+ release:
+    steps:
+    - git-clone:
+        title: "Clone Git repository."
+    - cache-pull:
+        title: "Pull cache."
+    - script:
+        title: "Run."
+        inputs:
+        - content: "bash ci/jobs/release.sh"
```

Comment:
nit: ideally, this line wouldn't be calling `bash` directly, script can define expected interpreter

Reply:
Also scripts don't need `.sh`, but that's subjective (no need to address, just commenting)